src/com/google/javascript/jscomp/CollapseProperties.java

```diff
@@ -90,21 +94,36 @@
 
   private final AbstractCompiler compiler;
   private final PropertyCollapseLevel propertyCollapseLevel;
+  private final ChunkOutputType chunkOutputType;
 
   /** Maps names (e.g. "a.b.c") to nodes in the global namespace tree */
   private Map<String, Name> nameMap;
 
   private final HashSet<String> dynamicallyImportedModules = new HashSet<>();
 
   CollapseProperties(AbstractCompiler compiler, PropertyCollapseLevel propertyCollapseLevel) {
+    this(compiler, propertyCollapseLevel, ChunkOutputType.GLOBAL_NAMESPACE);
+  }
+
+  CollapseProperties(
+      AbstractCompiler compiler,
+      PropertyCollapseLevel propertyCollapseLevel,
+      ChunkOutputType chunkOutputType) {
     this.compiler = compiler;
     this.propertyCollapseLevel = propertyCollapseLevel;
+    this.chunkOutputType = chunkOutputType;
   }
 
   @Override
   public void process(Node externs, Node root) {
-    if (propertyCollapseLevel == PropertyCollapseLevel.MODULE_EXPORT) {
-      gatherDynamicallyImportedModules();
+    if (propertyCollapseLevel == PropertyCollapseLevel.MODULE_EXPORT ||
+        chunkOutputType == ChunkOutputType.ES_MODULES) {
+      NodeTraversal.traverse(
+          compiler,
+          root,
+          new FindDynamicallyImportedModules(
+              compiler.getOptions().processCommonJSModules,
```

Comment:
This class shouldn't be directly reading Compiler options.
Please add constructor parameters as needed to avoid this.

Reply:
fixed